### PR TITLE
reusable workflow CI - checkout corrrect repository

### PR DESCRIPTION
**Pull request by mlee03**
**Description**
Currently, the CI checks out the incorrect repository and hash when the workflow is triggered externally.  This PR fixes this issue.

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included

### CODE REVIEW

File: .github/workflows/main_unit_tests_mpich.yaml [hunk 1]
```diff
@@ -1,26 +0,0 @@
-name: "Main unit tests with mpich"
```
Comment (mlee03) [inline .github/workflows/main_unit_tests_mpich.yaml]: this file must have snuck in somehow, it should not be in develop

File: .github/workflows/main_unit_tests_openmpi.yaml [hunk 2]
```diff
@@ -1,26 +0,0 @@
-name: "Main unit tests with openmpi"
```
Comment (mlee03) [inline .github/workflows/main_unit_tests_openmpi.yaml]: this file must have snuck in somehow, it should not be in develop

**Review by fmalatino — APPROVED** [review]
lgtm, probably should go in after [PR 86](https://github.com/NOAA-GFDL/pace/pull/86)

**Comment by mlee03** [comment]
@fmalatino thank you helping me test the new CI :smiley:

**Pull request merged**